### DISCUSSION TITLE
free memory

### DISCUSSION THREAD

**Discussion by vidilab** [arendst/Tasmota#17865] [Q&A]
We compile our own Tasmota version for our hardware, and it is getting a bit low on memory (~22kb), also considering we would like to be able to add more memory-using features later.
After upgrading from Tasmota 11.0.0 to the latest 12.3.1, free memory decreased with about 1 kb. Is it known where this 1kb is used for? I have looked through the changelog but not found it.
What is the best approach to find what to disable at compile time or other ways to free a bit more memory?

**Comment by Jason2866** [comment]
Features added. You find the changes in the changelog file, there are many changes and adds since 11.x. The easiest is to enable only the features which are needed. How to do is described in Tasmota documentation

**Reply by barbudor** [reply]
You can go through `user_config.h` to see what is enabled by default and then add the matching `#undef` in your `user_config_override.h`
Remove everything you feel you don't have a need.
My custom build "all-purpose" Tasmota leaves 25.2KB of free ram and use 564KB of flash and it's the binary I flash across various devices  so it includes drivers for everything I need.
